src/pages/day-view.tsx

```diff
@@ -0,0 +1,42 @@
+import type { NextPage } from "next";
+import { TodoItem } from "../features/todo/list";
+import { Title } from "../features/ui/title";
+import { trpc } from "../utils/trpc";
+
+const DayViewPage: NextPage = () => {
+    const inboxQuery = trpc.planning.inbox.useQuery();
+    const todos = inboxQuery.data?.todos;
+    const todoGroupedByGoal = todos?.reduce((acc, todo) => {
+        const goal = todo.goalName;
+        if (!acc[goal] || !Array.isArray(acc[goal])) {
+            acc[goal] = [];
+        }
+        else {
+            acc[goal]?.push(todo);
+        }
+        return acc;
+    }, {} as Record<string, typeof todos>);
+
+    if (todoGroupedByGoal === undefined) {
+        return <div>Loading...</div>;
+    }
+
+
+    return (
+        <main className="flex grow flex-col">
+            {todoGroupedByGoal && Object.entries(todoGroupedByGoal).map(([goal, todos]) => (
```

Comment:
The `todoGroupedByGoal` check is a bit redundant, since it's checked to have a value in the above if statement (type guard YES).

Also, It's better to use Lodash method `_.toPairs`, the result will be the same, but it doesn't throw error and crash the program like `Object.entries` does sometimes.